internal/otelmatchers/trace_matchers.go

```diff
@@ -0,0 +1,116 @@
+package otelmatchers
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+
+	"github.com/onsi/gomega"
+	"github.com/onsi/gomega/types"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+func ConsistOfSpansWithIDs(expectedSpanIDs []pcommon.SpanID) types.GomegaMatcher {
```

Comment:
I wonder if we can use go generics for both `ConsistOfSpansWithIDs` and `ConsistOfSpansWithIDs`?

Reply:
Please elaborate :)

Reply:
I would not say generics would solve it, we could instead introduce another helper to reduce code duplication, but not sure it's worth it